syntax="proto3";

option go_package="./pb";

package proto;

message Entry {
    string username = 1;
    string title = 2;
    string content = 3;
}

message RegisterEntryRequest {
    Entry entry = 1;
}

message RegisterEntryResponse {
    string message = 1;
}

message GetUserEntriesRequest {
    string username = 1;
}

message GetUserEntriesResponse {
    string message = 1;
    repeated Entry entries = 2;
}

message DeleteUserEntriesRequest {
    string username = 1;
}

message DeleteUserEntriesResponse {
    string message = 1;
}

service JournalService {
    rpc RegisterEntry(RegisterEntryRequest) returns (RegisterEntryResponse);
    rpc GetUserEntries(GetUserEntriesRequest) returns (GetUserEntriesResponse);
    rpc DeleteUserEntries(DeleteUserEntriesRequest) returns (DeleteUserEntriesResponse);
}